app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt

```diff
@@ -0,0 +1,137 @@
+      containerColor = MaterialTheme.colorScheme.background,
+      topBar = {
+        TopAppBar(
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("backButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Default.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            },
+            title = {
+              Text(text = "Confirmed Lesson", style = MaterialTheme.typography.titleLarge)
+            })
+      }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(horizontal = 16.dp)
+                    .verticalScroll(rememberScrollState())
+                    .testTag("confirmedLessonScreen"),
+            verticalArrangement = Arrangement.spacedBy(8.dp)) {
+              // Lesson Details Card
+              Card(
+                  modifier = Modifier.fillMaxWidth(),
+                  colors =
+                      CardDefaults.cardColors(
+                          containerColor = MaterialTheme.colorScheme.surfaceVariant)) {
+                    Column(
+                        modifier = Modifier.padding(8.dp),
+                        verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                          DisplayLessonDetails(lesson, otherProfile)
+
+                          LessonLocationDisplay(
+                              latitude = lesson.latitude,
+                              longitude = lesson.longitude,
+                              lessonTitle = lesson.title)
+                        }
+                  }
+
+              Spacer(modifier = Modifier.weight(1f))
+
+              // Contact Button
+              Button(
+                  shape = MaterialTheme.shapes.medium,
+                  onClick = {
+                    val intent =
+                        Intent(Intent.ACTION_VIEW).apply {
+                          data = Uri.parse("sms:${otherProfile.phoneNumber}")
+                          putExtra(
+                              "sms_body",
+                              "Hello, about our lesson ${formatDateTime(lesson.timeSlot)}...")
+                        }
+                    context.startActivity(intent)
+                  },
+                  modifier =
+                      Modifier.fillMaxWidth().padding(bottom = 16.dp).testTag("contactButton")) {
+                    Icon(
+                        Icons.Default.Send,
+                        contentDescription = null,
+                        modifier = Modifier.size(20.dp))
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text("Message ${if (isStudent) "Tutor" else "Student"}")
+                  }
+            }
+      }
+}
+
+private fun formatDateTime(timeSlot: String): String {
```

Comment:
I think this is a function that we have rewritten 1 or 2 times already, we should extract it in a specific file :) but this is not urgent.